stdlib/http/src/main/java/org/ballerinalang/net/http/websocket/WebSocketUtil.java

```diff
@@ -221,7 +251,482 @@ public static ErrorValue createErrorCause(String message, String reason, BPackag
+        RetryContext retryConnectorConfig = (RetryContext) webSocketClient.
+                getNativeData(WebSocketConstants.RETRY_CONFIG);
+        FailoverContext failoverConfig = (FailoverContext) webSocketClient.
+                getNativeData(WebSocketConstants.FAILOVER_CONFIG);
+        int interval = retryConnectorConfig.getInterval();
+        int maxInterval = retryConnectorConfig.getMaxInterval();
+        int maxAttempts = retryConnectorConfig.getMaxAttempts();
+        int noOfReconnectAttempts = retryConnectorConfig.getReconnectAttempts();
+        float backOfFactor = retryConnectorConfig.getBackOfFactor();
+        WebSocketService wsService = connectionInfo.getService();
+        Date date = new Date();
+        SimpleDateFormat formatter = new SimpleDateFormat("dd-MM-yyyy HH:mm:ss");
+        ArrayList targets = failoverConfig.getTargetUrls();
+        // Check the current url index equals with previous connected url index or not.
+        // If it is equal, update the no of reconnection attempt by one
+        if (currentIndex == failoverConfig.getInitialIndex()) {
+            logger.debug(LOG_MESSAGE, formatter.format(date.getTime()), WebSocketConstants.RECONNECTING);
+            noOfReconnectAttempts++;
+            retryConnectorConfig.setReconnectAttempts(noOfReconnectAttempts);
+        }
+        // Check the no of reconnection attempt equals with maximum reconnection attempt or not.
+        // If it isn't equal, call the initialiseWebSocketConnection()
+        // if it equals, return false
+        if (((noOfReconnectAttempts < maxAttempts) && maxAttempts > 0) || maxAttempts == 0) {
+            setCountDownLatch(getWaitTime(interval, maxInterval, backOfFactor, noOfReconnectAttempts));
+            establishFailoverConnection(createWebSocketClientConnector(targets.get(currentIndex).toString(),
+                    webSocketClient), webSocketClient, wsService);
+            return true;
+        }
+        logger.debug(WebSocketConstants.STATEMENT_FOR_FAILOVDER_RECONNECT + targets);
+        return false;
+    }
+
+    /**
+     * Set waiting time before attempting to next doReconnect/failover.
+     * @param interval interval
+     */
+    private static void setCountDownLatch(int interval) {
+        CountDownLatch countDownLatch = new CountDownLatch(1);
+        try {
+            if (!countDownLatch.await(interval, TimeUnit.MILLISECONDS)) {
+                countDownLatch.countDown();
+            }
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            throw new WebSocketException("Error occurred: " + e.getMessage());
+        }
+    }
+
+    public static boolean hasRetryConfig(ObjectValue webSocketClient) {
+        return webSocketClient.getMapValue(WebSocketConstants.CLIENT_ENDPOINT_CONFIG).
+                getMapValue(WebSocketConstants.RETRY_CONFIG) != null;
+    }
+
+    /**
+     * Calculate the waiting time.
```

Comment:
Let's leave a new line in all places
```suggestion
     *
     * Calculate the waiting time.
```